Update README to indicate docs submodule is archived (#246)
geonetwork/doc: README.md

# Archived GeoNetwork sphinx-build documentation module

> This repository conatains archived reStructured text documentation for GeoNetwork 4.2, 3.12, and 2.10 releases. 

Reviewer: ```suggestion
**This repository is archived and the changes won't be published in the GeoNetwork website anymore**

This repository contains archived reStructured text documentation for GeoNetwork 4.2, 3.12, and 2.10 releases. 
```

> The archived reStructuredText documentaiton is available here:

Reviewer: ```suggestion
The archived reStructuredText documentation is available here:
```

* https://geonetwork-opensource.org/manuals/

This content has migrated to markdown in the core-geonetwork repository:

* https://github.com/geonetwork/core-geonetwork/tree/main/docs/manual

## License

> This documentation is written under the creative commons license [Attribution-ShareAlike 3.0 Unported (CC BY-SA 3.0)](LICENSE.md).

Reviewer: ```suggestion
This documentation is written under the Creative Commons license [Attribution-ShareAlike 3.0 Unported (CC BY-SA 3.0)](LICENSE.md).
```
